.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,3 +1,10 @@
+#### What changes are you introducing?
+
+#### Why are you introducing these changes? (Explain or link to a Jira ticket)
```

Comment:
I generally don't like issues that don't have context and only link to some bug tracker. Can we separate those out into a References section? References can be to any bug tracker, related PRs (which can even be dependencies) or external documentation.

My aim would be to encourage people to always explain why they're doing something. Jira issues can describe a bug, but that can be a very long conversation. Rather than having the reviewer read up on the whole history it's easier if the PR author summarizes it.

Of course, for a trivial issue like a typo fix it's obvious and I don't expect anyone to fill it in.

Reply:
> I generally don't like issues that don't have context and only link to some bug tracker. Can we separate those out into a References section? References can be to any bug tracker, related PRs (which can even be dependencies) or external documentation.
> 
> My aim would be to encourage people to always explain why they're doing something. Jira issues can describe a bug, but that can be a very long conversation. Rather than having the reviewer read up on the whole history it's easier if the PR author summarizes it.

When I wrote line 3, I was again thinking of the art of the possible. What I too often see on other projects is that contributors don't even bother to fill out the PR template. That's very likely to happen here too, in which case giving the option to only link to an issue is a compromise: from a contributor's perspective, it's easy; from a reviewer's perspective, it's better than nothing.

> Of course, for a trivial issue like a typo fix it's obvious and I don't expect anyone to fill it in.

I understand what you mean but then again, everybody doesn't have the same understanding of what "trivial" means. A typo? Sure! But how about adding a command option? To some, it may be trivial too so they wouldn't be inclined to fill in the section; to me, it's not trivial at all because it's also a question of duplicating man pages or `--help` outputs and as a reviewer, I need to know what problem is actually being solved because that also affects where to place that option... So I would expect at least a word or two for every PR.

Reply:
> When I wrote line 3, I was again thinking of the art of the possible. What I too often see on other projects is that contributors don't even bother to fill out the PR template. That's very likely to happen here too, in which case giving the option to only link to an issue is a compromise: from a contributor's perspective, it's easy; from a reviewer's perspective, it's better than nothing.

My issue is with Jira. That's a Red Hat thing that upstream shouldn't really care about. It can be an interesting reference, but nothing upstream should care about. So if anything I'd go with something like this.
```suggestion
#### Why are you introducing these changes? (Explanation, links to references, issues, etc)
```

> I understand what you mean but then again, everybody doesn't have the same understanding of what "trivial" means. A typo? Sure! But how about adding a command option? To some, it may be trivial too so they wouldn't be inclined to fill in the section; to me, it's not trivial at all because it's also a question of duplicating man pages or `--help` outputs and as a reviewer, I need to know what problem is actually being solved because that also affects where to place that option... So I would expect at least a word or two for every PR.

I was trying to indicate that they fall in the SHOULD category, not MUST.